src/providers/gemini-provider.ts

```diff
@@ -0,0 +1,103 @@
+import { GoogleGenerativeAI, GenerativeModel, HarmCategory, HarmBlockThreshold } from '@google/generative-ai';
+import { LLMProvider } from './llm-provider';
+import { DefaultRequestBuilder, RequestBuilder } from './request-builder';
+import { handleUnknownError } from '../errors/index';
+
+export interface GeminiConfig {
+    apiKey: string;
+    model?: string;
+    temperature?: number;
+    debug?: boolean;
+    showPrompt?: boolean;
+    showPromptTrunc?: boolean;
+    debugJson?: boolean;
+}
+
+export class GeminiProvider implements LLMProvider {
+    private client: GoogleGenerativeAI;
+    private model: GenerativeModel;
+    private config: GeminiConfig;
+    private builder: RequestBuilder;
+
+    constructor(config: GeminiConfig, builder?: RequestBuilder) {
+        this.client = new GoogleGenerativeAI(config.apiKey);
+        this.config = {
+            ...config,
+            model: config.model ?? 'gemini-2.5-flash',
+            temperature: config.temperature ?? 0.2,
```

Comment:
Define these default values as constants, along with the default configuration for other models, and import them.